tests/testthat/test-prepare_pctr_product.R

```diff
@@ -98,3 +98,23 @@ test_that("handles tiltIndicator output", {
     )
   )
 })
+
+test_that("*matching_certainty_company_average column yield only 1 unique value", {
+
+  product_empty <- pctr_product |>
+    filter(clustered %in% c("building construction", "machining"))
+
+  result <- prepare_pctr_product(
+    product_empty,
+    ep_companies,
+    ecoinvent_activities,
+    small_matches_mapper,
+    isic_tilt_mapper
+  )
+  out <- result |>
+    group_by(companies_id) |>
+    summarise(count = n_distinct(matching_certainty_company_average))
+
+  expect_lte(unique(out$count), 1)
```

Comment:
@maurolepore I have created new toy datasets only for `company` and `product` datasets. If I have to create smaller datasets then I have to create datasets for `ep_companies`, `ecoinvent_activities`, `matches_mapper` as well because they have relevant data to create non-empty values in column `matching_certainty_company_average`. 
Please review!

Reply:
@maurolepore What's the reason behind using the `small_matches_mapper` ?

Reply:
@maurolepore What's the reason behind using the `small_matches_mapper` ?

Reply:
`small_matches_mapper` makes each call to `prepare_matches_mapper()` run about 50 times faster. Many tests end up calling `prepare_matches_mapper()` so the compound effect make the tests many many times faster. The faster the tests, the more often we'll run them -- ideally after each tiny change -- and the faster we'll catch mistakes.

``` r
devtools::load_all()
#> ℹ Loading tiltIndicatorAfter

system.time(prepare_matches_mapper(small_matches_mapper, ecoinvent_activities))
#>    user  system elapsed 
#>   0.033   0.001   0.034

# ~ 50 times slower
system.time(prepare_matches_mapper(matches_mapper, ecoinvent_activities))
#>    user  system elapsed 
#>   1.629   0.028   1.657
```

<sup>Created on 2023-10-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
I know it's a pain to create toy data but it's also super important. It's time we spend in the beginning of a package, but then it saves a ton of time by allowing us to understand and run tests faster, which eventually translates to testing the code more completely and saving time that would otherwise be spend fixing bugs.

Reply:
@maurolepore Indeed I agree! I am using `small_matches_mapper` now to run tests faster. I was able to find one example which is able to match all the tables for the tests I am working on. I hope now all the tests are working on the smallest datasets possible for the specific usecases I am testing. Please review!